stdio/perror.c

```diff
@@ -17,7 +17,12 @@
 #include <errno.h>
 #include <string.h>
 
+
 void perror(const char *str)
 {
-	fprintf(stderr, "%s: %s\n", str, strerror(errno));
+	char errstr[64];
+
+	(void)strerror_r(errno, errstr, sizeof(errstr));
+
+	fprintf(stderr, "%s: %s\n", str, errstr);
```

Comment:
I suggest adding this:
```suggestion
	if ((str != NULL) && (*str != '\0')) {
		fprintf(stderr, "%s: ", str);
	}
	fprintf(stderr, "%s\n", errstr);
```

Reply:
I've applied only NULL check, IMHO no need to check for empty string, it doesn't hurt anything